### PR TITLE
QuestDB formula

### PR DESCRIPTION
**Pull request by bluestreak01**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

New formula for brew installation of QuestDB

### CODE REVIEW

File: Formula/questdb.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
+class Questdb < Formula
+  desc "Time Series Database"
+  homepage "https://www.questdb.org"
+  url "https://www.questdb.org/download/questdb-1.0.0-SNAPSHOT-20161013-1304-bin.tar.gz"
```
Comment (MikeMcQuaid) [inline Formula/questdb.rb]: This doesn't look like a stable release: https://github.com/bluestreak01/questdb/releases. We need to use one of those.

File: Formula/questdb.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  sha256 "66f4d43868e8d0eb61606712a25f397023c82e368bbcbfda7cc007611661984b"
+  bottle :unneeded
```
Comment (MikeMcQuaid) [inline Formula/questdb.rb]: Add a newline above this line.

File: Formula/questdb.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  depends_on :java => "1.7+"
```
Comment (MikeMcQuaid) [inline Formula/questdb.rb]: Add a newline above this line.

File: Formula/questdb.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+  def datadir
+    var/"questdb"
```
Comment (MikeMcQuaid) [inline Formula/questdb.rb]: This is only used in one place so hardcode it instead.

File: Formula/questdb.rb [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  end
+
+  def install
+    rm_rf "questdb.exe"
+    prefix.install Dir["*"]
```
Comment (MikeMcQuaid) [inline Formula/questdb.rb]: Might be worth using `libexec` instead; that's what we generally do.

File: Formula/questdb.rb [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+          <string>#{opt_bin}/questdb</string>
+          <string>start</string>
+          <string>-d</string>
+          <string>#{datadir}</string>
+          <string>-n</string>
+        </array>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>WorkingDirectory</key>
+        <string>#{var}</string>
```
Comment (MikeMcQuaid) [inline Formula/questdb.rb]: Feels like this should be a subdirectory or not specified.

File: Formula/questdb.rb [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+          <integer>1024</integer>
+        </dict>
+      </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    system "#{bin}/questdb", "start"
+    sleep 2
```
Comment (MikeMcQuaid) [inline Formula/questdb.rb]: Add a test to verify you can e.g. connect to the process and an `ensure` to kill it if anything fails.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by bluestreak01** [comment]
Thanks a lot for the review Mike, I'll make suggested changes are re-submit.

Which directory can I use on osx to store data. I noticed that other projects use /usr/local/var and it worked on my osx, but jenkins failed with permission problems. Would you be able to advise?

File: Formula/questdb.rb (hunk 1 above)
Comment (bluestreak01) [inline Formula/questdb.rb]: Can we let this pull thru with snapshot version? I realise we need a release, but i'm not quite ready with all of the testing.

Comment (MikeMcQuaid) [inline Formula/questdb.rb]: We cannot, sorry.

Comment (bluestreak01) [inline Formula/questdb.rb]: included release

File: Formula/questdb.rb (hunk 2 above)
Comment (bluestreak01) [inline Formula/questdb.rb]: done

File: Formula/questdb.rb (hunk 3 above)
Comment (bluestreak01) [inline Formula/questdb.rb]: done

File: Formula/questdb.rb (hunk 5 above)
Comment (bluestreak01) [inline Formula/questdb.rb]: done

File: Formula/questdb.rb (hunk 6 above)
Comment (bluestreak01) [inline Formula/questdb.rb]: fixed with non-generic directory

**Pull request closed without merging**